### PR TITLE
Update analyzer to 1.0.0.

### PR DESCRIPTION
**Pull request by jcollins-g**
This sets the minimum constraint for analyzer on relevant packages to 1.0.0, and deals with the source_gen problem by commenting out json_serializable.

Also switches testing back to the dev branch, as this will now work with the new analyzer.

### CODE REVIEW

File: build_resolvers/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 1.5.4
```
Comment (jakemac53) [inline build_resolvers/CHANGELOG.md:1]:
```suggestion
## 1.5.4-dev
```

**Review by jakemac53 — APPROVED** [review]

File: build_resolvers/CHANGELOG.md (hunk 1 above)
Comment (jcollins-g) [reply build_resolvers/CHANGELOG.md]: done

**Comment by jakemac53** [comment]
@jcollins-g looks like we need another override on `build_config` in the test at `build_runner/test/generate/run_script_test.dart`

**Comment by jcollins-g** [comment]
@jakemac53 yep, just figured that out myself.  Trying again.  Oddly enough, the problem won't reproduce locally.

**Comment by jakemac53** [comment]
> @jakemac53 yep, just figured that out myself. Trying again. Oddly enough, the problem won't reproduce locally.

I think these tests don't run by default locally, they are tagged as "integration" tests and we exclude those by default locally as they take forever to run, you can do `-t integration` to run them.

**Comment by jcollins-g** [comment]
> > @jakemac53 yep, just figured that out myself. Trying again. Oddly enough, the problem won't reproduce locally.
> 
> I think these tests don't run by default locally, they are tagged as "integration" tests and we exclude those by default locally as they take forever to run, you can do `-t integration` to run them.

No, I was running them, just couldn't reproduce the problem.

**Comment by jcollins-g** [comment]
If this does manage to pass, the PR should be mergeable.  PTAL.

**Comment by jcollins-g** [comment]
merging per offline discussion

**Pull request merged**